Assignment3/com/hotWax/Assignment3/EmployeeDAO.java

```diff
@@ -0,0 +1,245 @@
+        List<Employee> tempList = createArrayList(array);
+        boolean flag=false;
+        for(int i=0; i<tempList.size(); i++)
+        {
+            if(tempList.get(i).getEmployeeId().equals(String.valueOf(employeeId))) {
+                tempList.remove(i);
+                flag = true;
+                randomAccessFile.close();
+                file.delete();
+                break;
+            }
+        }
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        if(flag==true){System.out.println("employee deleted successfully");}
+        file = new File("Employees.txt");
+        randomAccessFile = new RandomAccessFile(file,"rw");
+        for(int i=0; i<tempList.size(); i++)
+        {
+            randomAccessFile.writeBytes(tempList.get(i).getEmployeeId()+",");
+            randomAccessFile.writeBytes(tempList.get(i).getName()+",");
+            randomAccessFile.writeBytes(simpleDateFormat.format(tempList.get(i).getDateOfBirth())+",");
+            randomAccessFile.writeBytes(tempList.get(i).getEmailId()+",");
+            randomAccessFile.writeBytes(String.valueOf(tempList.get(i).getAge())+",");
+        }
+        randomAccessFile.close();
+
+    }
+    public void searchByEmployeeId(int employeeId) throws DAOException, IOException {
+        if(employeeId<=0){
+            throw new DAOException("Invalid employee id.");
+        }
+         file = new File("Employees.txt");
+        if(!file.exists()) throw new DAOException("file not exist");
+        RandomAccessFile randomAccessFile = new RandomAccessFile(file,"r");
+        String fString = randomAccessFile.readLine();
+        String array[] = fString.split(",");
+        List<Employee> tempList = createArrayList(array);
+        for(int i=0; i<tempList.size(); i++)
+        {
+            if(tempList.get(i).getEmployeeId().equals(String.valueOf(employeeId))) {
+                System.out.print(tempList.get(i).getEmployeeId()+" , ");
+                System.out.print(tempList.get(i).getName()+" , ");
+                System.out.print(tempList.get(i).getDateOfBirth()+" , ");
+                System.out.print(tempList.get(i).getEmailId()+" , ");
+                System.out.println(tempList.get(i).getAge());
+            }
+        }
+        randomAccessFile.close();
+
+    }
+
+    public void searchByName(String name) throws DAOException, IOException {
+        if(name==null){
+            throw new DAOException("Invalid name");
+        }
+        file = new File("Employees.txt");
+        if(!file.exists()) throw new DAOException("file not exist");
+        RandomAccessFile randomAccessFile = new RandomAccessFile(file,"r");
+        String fString = randomAccessFile.readLine();
+        String array[] = fString.split(",");
+        List<Employee> tempList = createArrayList(array);
+        for(int i = 0; i<tempList.size(); i++)
+        {
+            if(tempList.get(i).getName().equalsIgnoreCase(name)) {
```

Comment:
same here if you found the record then you can end the loop otherwise this will iterate the complete list